src/webhook/__tests__/index.js

```diff
@@ -0,0 +1,66 @@
+jest.mock('src/webhook/checkSignatureAndParse');
+jest.mock('src/lib/redisClient');
+
+import Koa from 'koa';
+import request from 'supertest';
+import MockDate from 'mockdate';
+
+import webhookRouter from '../';
+import UserSettings from '../../database/models/userSettings';
+import Client from '../../database/mongoClient';
+
+describe('Webhook router', () => {
+  beforeAll(async () => {
+    MockDate.set(612921600000);
+
+    if (await UserSettings.collectionExists()) {
+      await (await UserSettings.client).drop();
+    }
+  });
+
+  afterAll(async () => {
+    await (await Client.getInstance()).close();
+    MockDate.reset();
+  });
+
+  it('singleUserHandler() should handle follow event', async () => {
+    const app = new Koa();
+    app.use(webhookRouter.routes(), webhookRouter.allowedMethods());
+
+    const eventObject = {
+      events: [
+        {
+          replyToken: 'nHuyWiB7yP5Zw52FIkcQobQuGDXCTA',
+          type: 'follow',
+          mode: 'active',
+          timestamp: 1462629479859,
+          source: {
+            type: 'user',
+            userId: 'U4af4980629',
+          },
+        },
+      ],
+    };
+
+    const server = app.listen();
+
+    await request(server)
```

Comment:
Since you invoke `UserSettings.findOrInsertByUserId` directly in the test script (L52), this test will pass even if you comment out L47-L50.

In this test, I suggest we read from DB using MongoDb commands, instead of invoking `UserSettings.findOrInsertByUserId`.